### PR TITLE
Dependency Check

### PR DESCRIPTION
**Pull request by originalbchan**

### CODE REVIEW

**Comment by kconwayatlassian** [comment]
question: How do you plan to make this available to callers? I don't see an example in the sample OpenAPI specification.

File: pkg/dependencycheck/dependencycheck.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	req, _ := http.NewRequest(http.MethodGet, u.String(), http.NoBody)
+	res, err := dc.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != 200 {
+		// Nexpose is down?  Nexpose unexpectedly gave 404?
+		headers, errMarshal := json.Marshal(res.Header)
```
Comment (kconwayatlassian) [inline pkg/dependencycheck/dependencycheck.go]: question: Why are we logging response headers? What information are we attempting to capture?

Comment (originalbchan) [reply pkg/dependencycheck/dependencycheck.go]: yea that was some copy pasta, i'll remove it

File: pkg/dependencycheck/dependencycheckconfig.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+// and make a call to the fetch assets API
+type DependencyCheckConfig struct {
+	HTTPClient      *httpclient.Config `description:"The HTTP client config from github.com/asecurityteam/component-httpclient."`
+	NexposeEndPoint string             `description:"The scheme and host of a Nexpose instance."`
+}
+
+// Name is used by the settings library and will add a "NEXPOSE_"
+// prefix to NexposeConfig environment variables
+func (c *DependencyCheckConfig) Name() string {
+	return "NexposeDependencyCheck"
```
Comment (originalbchan) [inline pkg/dependencycheck/dependencycheckconfig.go]: check environment variables and config: call name DependencyCheck

File: pkg/dependencycheck/dependencycheck.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+// DependencyCheck implements the interfaces to fetch vulnerabilities and solutions from nexpose
+type DependencyCheck struct {
+	HTTPClient      *http.Client
+	NexposeEndpoint *url.URL
+}
+
+// CheckDependencies calls a Nexpose endpoint and expects a 200
+func (dc *DependencyCheck) CheckDependencies(ctx context.Context) error {
```
Comment (LorenBoren) [inline pkg/dependencycheck/dependencycheck.go]: Question: Is the intent that in the future if this service starts calling other dependencies, that their checks will also go here in this function?

File: pkg/handlers/v1/dependencycheck.go [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package v1
+
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-vuln-hydrator/pkg/domain"
+)
+
+// DependencyCheckHandler is cool
```
Comment (LorenBoren) [inline pkg/handlers/v1/dependencycheck.go]: Task: Yeah, no, plz update

File: pkg/dependencycheck/dependencycheck.go (hunk 3 above)
Comment (originalbchan) [reply pkg/dependencycheck/dependencycheck.go]: yes

**Pull request merged**